### DISCUSSION TITLE
Hope to support object mode for `twin.config`

### DISCUSSION THREAD

**Discussion by targeral** [ben-rogerson/twin.macro#715] [Ideas]
In some scenarios, it is desirable to support the following：
``` js
// frame-babel-preset.js

// This configuration comes from a combination of other configurations provided by the framework
const configIsCombinationOfMultipleConfigurations = { /**/ };

module.exports = {
   plugins: [
      'babel-plugin-macros',
       {
           twin: { preset: 'styled-components', config:  configIsCombinationOfMultipleConfigurations }
       }
   ]
};
```

**Comment by ben-rogerson** [comment]
Dynamic babel configs should be already possible, what part are you running into trouble with?

```js
// babel-plugin-macros.config.js

const path = require('path');

const options = {
    preset: 'styled-components',
    debugPlugins: false,
};

module.exports = {
    twin: {
        config: path.resolve(__dirname, 'tailwind.config.js'),
        ...options,
    },
};
```

**Reply by ben-rogerson** [reply]
btw this is a babel thing rather than a twin thing.

**Reply by targeral** [reply]
Our framework's support for tailwindcss is through custom configuration rather than by reading the tailwindcss.config.js file, and there are two api's associated with configuring tailwindcss in our framework, It probably looks like this:

``` js
// frame.config.js
module.exports = {
     tools: {
        tailwindcss: {},
     },
     source: {
        designSystem: {} // This configuration is equivalent to the theme configuration of tailwindcss
     }
}
```

I understand from reading the source code of twin.macro that twin.macro reads the `twin.config` on the babel configuration as a path only for parsing and then reads the path to the specific tailwindcss configuration object. I would like to support passing in a tailwindcss configuration object directly.

**Reply by ben-rogerson** [reply]
Regarding twin config: From the babel file (`babel-plugin-macros.config.js`), have you tried requiring your config?

Eg: `const frameConfig = require('path/to/your/frame.config.js')`

And regarding the tailwindcss config - I like the sound of that feature - would you like to provide a PR?

**Comment by ben-rogerson** [comment]
In the latest twin rc, you can now pass a tailwind config object directly in via the `config` option in the twin config 🎉 
I've added some more info [in the docs](https://github.com/ben-rogerson/twin.macro/blob/master/docs/options.md#options)
